### PR TITLE
#306 Member does not see the description of the Userstory

### PR DESCRIPTION
**Pull request by SponsoredByPuma**
Its currently working, but there are still some error in the console, because of the changed key of the v-for iteration.
We probably could fix this by giving the userstory another attribute e.g. position, so we could iterate with this.
Here is a screenshot of the error message: 
image

### CODE REVIEW

File: frontend/src/components/MobileStoryList.vue [hunk 1]
```diff
@@ -2,7 +2,7 @@
   <div>
     <b-list-group-item
       v-for="(story, index) of userStories"
-      :key="story.name"
+      :key="story.position"
```
Comment (KzuDemEvin) [inline frontend/src/components/MobileStoryList.vue:5]: Why don't you use the property `id` of userStories or use the `index` as key?

File: frontend/src/components/MobileStoryTitle.vue [hunk 2]
```diff
@@ -4,7 +4,7 @@
       <b-list-group-item
         v-for="(story, idx) of userStories"
         v-show="story.isActive"
-        :key="story.name"
+        :key="story.position"
```
Comment (KzuDemEvin) [inline frontend/src/components/MobileStoryTitle.vue:7]: Why don't you use the property `id` of userStories or use the index `idx` as key?

File: frontend/src/components/UserStoryDescriptions.vue [hunk 3]
```diff
@@ -4,7 +4,7 @@
       <b-list-group-item
         v-for="(story, idx) of userStories"
         v-show="idx === index"
-        :key="story.name"
+        :key="story.position"
```
Comment (KzuDemEvin) [inline frontend/src/components/UserStoryDescriptions.vue:7]: Why don't you use the property `id` of userStories or use the index `idx` as key?

File: frontend/src/components/UserStoryDescriptions.vue [hunk 4]
```diff
@@ -95,43 +99,50 @@ export default Vue.extend({
...
       description: string;
       estimation: string | null;
       isActive: boolean;
+      position: number;
     }>;
   },
   methods: {
+    valueChanged(idx, {markdown}) {
+      this.userStories[idx].description = markdown;
+    },
```
Comment (KzuDemEvin) [inline frontend/src/components/UserStoryDescriptions.vue:124]: Perhaps you can already call `this.publishChanges(index);` here, so there is no need to press the button to update the user story for all members.

**Review by KzuDemEvin — CHANGES_REQUESTED** [review]
- Is the position really needed as a property? Or can it be replaced by the id of the user stories or the index in the loop?
- I had a suggestion regarding the update policy for user stories.

**Comment by SponsoredByPuma** [comment]
Thanks for the advice @KzuDemEvin. I removed the position attribute and added id as the key. Also I extended the mobile classes for the id attribute since it was missing in those. Last but not least the description now updates as soon as you click next to the markdown editor box.

**Review by KzuDemEvin — APPROVED** [review]

**Pull request merged**